rslearn/tile_stores/default.py

```diff
@@ -0,0 +1,332 @@
+)
+
+from .tile_store import TileStore
+
+# Special filename to indicate writing is done.
+COMPLETED_FNAME = "completed"
+
+# Filename to use for vector data.
+VECTOR_FNAME = "data.geojson"
+
+
+class DefaultTileStore(TileStore):
+    """Default TileStore implementation.
+
+    It stores raster and vector data under the provided UPath.
+    """
+
+    def __init__(
+        self,
+        path_suffix: str = "tiles",
+        convert_rasters_to_cogs: bool = True,
+        tile_size: int = 256,
+    ):
+        """Create a new DefaultTileStore.
+
+        Args:
+            path_suffix: the path suffix to store files under, which is joined with
+                the dataset path if it does not contain a protocol string. See
+                rslearn.utils.fsspec.join_upath.
+            convert_rasters_to_cogs: whether to re-encode all raster files to tiled
+                GeoTIFFs.
+            tile_size: if converting to COGs, the tile size to use.
+        """
+        self.path_suffix = path_suffix
+        self.convert_rasters_to_cogs = convert_rasters_to_cogs
+        self.tile_size = tile_size
+
+        self.path: UPath | None = None
+
+    def set_dataset_path(self, ds_path: UPath) -> None:
+        """Set the dataset path.
+
+        Args:
+            ds_path: the dataset path.
+        """
+        self.path = join_upath(ds_path, self.path_suffix)
+
+    def _get_raster_dir(
+        self, layer_name: str, item_name: str, bands: list[str]
+    ) -> UPath:
+        assert self.path is not None
+        return self.path / layer_name / item_name / "_".join(bands)
+
+    def is_raster_ready(
+        self, layer_name: str, item_name: str, bands: list[str]
+    ) -> bool:
+        """Checks if this raster has been written to the store.
+
+        Args:
+            layer_name: the layer name or alias.
+            item_name: the item.
+            bands: the list of bands identifying which specific raster to read.
+
+        Returns:
+            whether there is a raster in the store matching the source, item, and
+                bands.
+        """
+        raster_dir = self._get_raster_dir(layer_name, item_name, bands)
+        return (raster_dir / COMPLETED_FNAME).exists()
+
+    def get_raster_bands(self, layer_name: str, item_name: str) -> list[list[str]]:
+        """Get the sets of bands that have been stored for the specified item.
+
+        Args:
+            layer_name: the layer name or alias.
+            item_name: the item.
+
+        Returns:
+            a list of lists of bands that are in the tile store (with one raster
+                stored corresponding to each inner list).
+        """
+        assert isinstance(self.path, UPath)
+        item_dir = self.path / layer_name / item_name
+        if not item_dir.exists():
+            return []
+
+        bands: list[list[str]] = []
+        for raster_dir in item_dir.iterdir():
+            parts = raster_dir.name.split("_")
+            bands.append(parts)
```

Comment:
I think we can reserve "_" so it is not allowed in the band name. And then for data sources where there is underlying band called e.g. "lai_lv" it can rename it e.g. "lai-lv".

Reply:
I added throwing ValueError if one of the provided bands contains underscore.

Reply:
Sounds good!